src/main/java/com/company/demoapplication/TaskPublisher.java

```diff
@@ -55,7 +55,7 @@ private static SendMessageRequest createRequest(String sqsQueueUrl, String key)
     }
 
     private static List<String> listImagesOnS3() {
-        List<S3ObjectSummary> images = Main.s3Client().listObjectsV2(new ListObjectsV2Request().withBucketName(Main.bucketName).withPrefix(Main.sampleImagesFolder)).getObjectSummaries();
+        List<S3ObjectSummary> images = Main.s3Client().listObjects(new ListObjectsRequest().withBucketName(Main.bucketName).withPrefix(Main.sampleImagesFolder)).getObjectSummaries();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).